src/haz3lcore/statics/Statics.re

```diff
@@ -297,6 +300,7 @@ and upat_to_info_map =
       ~is_synswitch,
       ~ctx,
       ~ancestors: Info.ancestors,
+      ~free: Ctx.co=[],
```

Comment:
at least as an implementation strategy, i suggest removing the default value of this parameter (yes, i remember that i told you to add it :) to make sure you examine all the use sites (e.g. the logic you've added for the let should be added for case & lambda as well). if there aren't that many call sites i would suggest just leaving it non-optional and specifying an empty free for callsites that would use the default value... your call, this is a tradeoff